backend/bang-ggood/src/main/java/com/bang_ggood/auth/service/AuthService.java

```diff
@@ -30,20 +32,49 @@ public class AuthService {
     private final JwtTokenProvider jwtTokenProvider;
     private final JwtTokenResolver jwtTokenResolver;
     private final DefaultChecklistService defaultChecklistService;
-    private final UserRepository userRepository; // TODO 리팩토링
+    private final PasswordEncoder passwordEncoder;
+    private final UserRepository userRepository;
+
+    private static void validateTokenOwnership(User user, AuthUser accessAuthUser, AuthUser refreshAuthUser) {
+        if (!accessAuthUser.id().equals(refreshAuthUser.id())) {
+            throw new BangggoodException(ExceptionCode.AUTHENTICATION_TOKEN_USER_MISMATCH);
+        }
+        if (!user.getId().equals(accessAuthUser.id())) {
+            throw new BangggoodException(ExceptionCode.AUTHENTICATION_TOKEN_NOT_OWNED_BY_USER);
+        }
+    }
 
     @Transactional
-    public AuthTokenResponse login(OauthLoginRequest request) {
+    public AuthTokenResponse authLogin(OauthLoginRequest request) {
         OauthInfoApiResponse oauthInfoApiResponse = oauthClient.requestOauthInfo(request);
 
-        User user = userRepository.findByEmail(oauthInfoApiResponse.kakao_account().email())
+        User user = userRepository.findByEmailAndLoginType(oauthInfoApiResponse.kakao_account().email(),
+                        LoginType.KAKAO)
                 .orElseGet(() -> signUp(oauthInfoApiResponse));
 
         String accessToken = jwtTokenProvider.createAccessToken(user);
         String refreshToken = jwtTokenProvider.createRefreshToken(user);
         return AuthTokenResponse.of(accessToken, refreshToken);
     }
 
+    @Transactional(readOnly = true)
+    public AuthTokenResponse localLogin(LocalLoginRequestV1 request) {
+        User user = userRepository.findByEmailAndLoginType(request.email(), LoginType.LOCAL)
+                .orElseThrow(() -> new BangggoodException(ExceptionCode.USER_NOT_FOUND));
+        checkPassword(request, user);
+
+        String accessToken = jwtTokenProvider.createAccessToken(user);
+        String refreshToken = jwtTokenProvider.createRefreshToken(user);
+        return AuthTokenResponse.of(accessToken, refreshToken);
+    }
+
+    private void checkPassword(LocalLoginRequestV1 request, User user) {
```

Comment:
```suggestion
    private void validatePassword(LocalLoginRequestV1 request, User user) {
```
지금까지는 검증 메서드를 validate라고 했던 것 같아서 제안드려봅니다~!

Reply:
해당 부분은 정책에 따라 규칙에 맞는가 확인하는 것이 아닌 맞는가 체크하는 부분이라 생각해서 validate 명명보다는 비즈니스 로직 명명을 가질 수 있도록 구현하였습니다!

Reply:
제가 이해한 바에 따르면, 단순히 비밀번호가 다른지 체크하는 부분은 정책적인 부분이 아니라 네이밍을 다르게 가져가고 싶다는 의미로 이해했어요! 하지만 null, empty 체크도 비슷한 경우라고 생각되는데 validate 메서드명을 사용했던 것 같아서 이 부분은 어떻게 생각하는지 궁금합니다 ~

Reply:
null 혹은 empty 체크는 값 자체의 형식에 대한 검증이라 생각해요!
validate 검증 로직은 주로 입력 데이터가 특정 형식, 패턴 또는 규칙에 부합하는지 확인하는 데 사용하는 명명이라는 생각입니다
반면에 비밀번호 일치 여부를 확인하는 로직은 단순히 입력 데이터의 형식 검증이 아닌, 기존 데이터와 비교하여 중요한 비즈니스 결정을 내리는 과정이라 생각했어요!
validate라는 검증의 의미만 담기에는 다소 의미가 축소되는 느낌이 있다고 판단하였습니다